src/event-processor/event-processor.service.ts

```diff
@@ -0,0 +1,223 @@
+import {
+  Injectable,
+  Logger,
+  OnModuleInit,
+  OnApplicationShutdown,
+} from '@nestjs/common';
+import { ConfigService } from '@nestjs/config';
+import { Cron } from '@nestjs/schedule';
+import * as amqp from 'amqplib';
+import { ConnectorService } from './connector.service';
+import { UpdatePaymentStatusDto } from './dto/update-payment-status.dto';
+import { PaymentStatus } from 'src/shared/enums/payment-status.enum';
+import { ShipmentStatus } from 'src/shared/enums/shipment-status.enum';
+import { UpdateShipmentStatusDto } from './dto/update-shipment-status';
+// import { ConnectionService } from './connection.service'; // Adjust import path as needed
+
+/**
+ * The service to process events from the respective queues.
+ * @property processingTime The upper limit of simulated processing time
+ * @property connection The current RabbitMQ connection
+ * @property queues An array containing all queue identifiers, to connect to
+ * @property maxMessages The rate limit for each queue per minute
+ * @property messageCounts The current message count
+ * @property booolean flag
+ */
+@Injectable()
+export class EventProcessorService
+  implements OnModuleInit, OnApplicationShutdown
+{
+  private readonly processingTime: number;
+  private connection: amqp.Connection;
+  private channel: amqp.Channel;
+  private queues = ['payments-queue', 'shipments-queue'];
+  private maxMessages: number;
+  private successRate: number;
+  private messageCounts: { [key: string]: number } = {};
+  private processingAllowed: { [key: string]: boolean } = {};
+
+  constructor(
+    private readonly logger: Logger,
+    private configService: ConfigService,
+    private connectorService: ConnectorService,
+  ) {
+    // read configuration from environment variables
+    this.processingTime = this.configService.get<number>(
+      'PROCESSING_TIME_SECONDS',
+      5000,
+    );
+    this.maxMessages = this.configService.get<number>(
+      'PAYMENTS_PER_MINUTE',
+      100,
+    );
+    this.successRate = this.configService.get<number>('SUCCESS_RATE', 0.95);
+
+    // setup queue variables
+    this.queues.forEach((queue) => {
+      this.messageCounts[queue] = 0;
+      this.processingAllowed[queue] = true;
+    });
+  }
+
+  async onModuleInit() {
+    let rabbitMQConnected: boolean = await this.connectToRabbitMQ();
+
+    while (!rabbitMQConnected) {
+      this.logger.debug('Retrying in 5 seconds.');
+      await new Promise((resolve) => setTimeout(resolve, 5000));
+      rabbitMQConnected = await this.connectToRabbitMQ();
+    }
+    this.queues.forEach((queue) => this.initializeConsumer(queue));
+  }
+
+  /**
+   * connects to the RabbitMQ Queue
+   * @returns boolean that indicates if the connection was successful
+   */
+  private async connectToRabbitMQ(): Promise<boolean> {
+    const rabbitmqUrl = this.configService.get<string>(
+      'RABBITMQ_URL',
+      'amqp://localhost',
```

Comment:
is this a reasonable fallback?

Reply:
TBH there does not need to be an fallback